Level 3:

Task 1: 
- Analyze the text reviews to identify the most common positive and negative keywords.
- Calculate the average length of reviews and explore if there is a relationship between review length and rating.

In [36]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns

# Load the data
df = pd.read_csv('/Users/tilakpatel/Personel/Cognifyz intern/code/Data/Dataset.csv', encoding='utf-8')
df = df.dropna(subset=['Cuisines'])

In [37]:
df_reviews = df.copy()

In [38]:
df_reviews['rating length'] = df_reviews['Rating text'].apply(lambda x: len(x.split()))

In [45]:
df_length_rating = pd.DataFrame(df_reviews.groupby('rating length')['Aggregate rating'].mean()).reset_index()

In [49]:
df_length_rating.rename(columns={'rating length': 'rating length (no. of words)'}, inplace=True)

In [50]:
df_length_rating

rating length (no. of words)  Aggregate rating
0                             1          3.315184
1                             2          1.392746

- Shorter reviews (1 word) have a higher average rating (3.32), suggesting they are more positive. 
- Longer reviews (2 words) have a lower average rating (1.39), implying they are more critical or

In [42]:
df_reviews['Rating text'].value_counts()

Rating text
Average      3734
Not rated    2148
Good         2096
Very Good    1078
Excellent     300
Poor          186
Name: count, dtype: int64

- Positive Sentiment: The words "Excellent," "Very Good," and "Good" are the positive keywords, with a significant count of reviews reflecting positive feedback.

- Negative Sentiment: "Poor" and "Not rated" indicate negative or neutral feedback, with "Not rated" having the highest count.